### PR TITLE
feat(Resolution): 1-Add ResolveMultiple

### PR DESCRIPTION
**Pull request by yparkTaBS**

### CODE REVIEW

File: pkg/GoFac/GoFac_test.go [hunk 1]
```diff
@@ -154,3 +154,87 @@ GoFac.Resolve: github.com/TaBS/GoFac/tests/SampleStructs/IIndependentStruct is n
...
+		func() {
+			result, err = Resolve[ss.IStructRelyingOnIndependentStruct](container)
+		}, 
+		"Should not have paniced when resolving interface!",
+	)
+
+	assert.NotNil(result, "Resolved object should not be nil!")
+	assert.Nil(err, "Should not have any error!")
+	assert.Equal("StructRelyingOnIndependentStructs", result.ReturnNameStructRelyingOnIndependentStruct(), "Names of the struct is different!")
+	assert.Contains(result.(ss.IStructRelyingOnIndependentStructs).ReturnSubStructNames(), "IndependentStruct", "IndependentStruct should have been resolved too!")
```
Comment (yparkTaBS) [inline pkg/GoFac/GoFac_test.go:198]: check whether there are only two items

File: pkg/GoFac/GoFac.go [hunk 2]
```diff
@@ -49,10 +49,10 @@ func Resolve[T interface{}](container *Container) (T, error) {
 
 	tInfo := reflect.TypeFor[T]()
 	if isArrayOrSlice(tInfo) {
-		tInfo = tInfo.Elem()
+		return base, h.MakeError("GoFac.Resolve", "Use ResolveMultiple for resolving an array or a slice")
```
Comment (yparkTaBS) [inline pkg/GoFac/GoFac.go:52]: Check whether the T is other types or not.

File: pkg/GoFac/GoFac.go [hunk 3]
```diff
@@ -61,14 +61,41 @@ func Resolve[T interface{}](container *Container) (T, error) {
...
+		return base, err
+	}
+
+	if !isValueArrayOrSlice(*dependency) {
+		return base, h.MakeError("GoFac.Resolve", "Resulting dependency is not array, though it needed to be! Something went horribly wrong!")
+	}
+
+	var resolutions []T
+	for i := 0; i < (*dependency).Len(); i++ {
+		resolution, ok := (*dependency).Index(i).Interface().(reflect.Value).Interface().(T)
```
Comment (yparkTaBS) [inline pkg/GoFac/GoFac.go:86]: Comment would be helpful.

**Review by mokimokheonpark — APPROVED** [review]
Looks good.

**Review by mokimokheonpark — APPROVED** [review]
Looks good.

**Pull request merged**